documentation/src/docs/controller-service.md

```diff
@@ -374,34 +379,19 @@ value is updated during the course of the operation, it is again
 
-## Stream Buckets
-
-To enable some scenarios, we may need the background workers to
-periodically work on each of the Streams in our cluster to perform
-some specific action on them. The concept of Stream Bucket is to
-distribute this periodic background work across all available
-Controller instances. Controller instances map all available streams in the system into buckets and these buckets are distributed amongst themselves. Hence, all the long-running background work can be uniformly distributed across multiple Controller instances.
-
-**Note**: Number of buckets for a cluster is a fixed (configurable) value for
-the lifetime of a cluster.
+### Key Value Table Metadata Store
+Just like stream metadata store, pravega key value tables have their own metadata which is accessed using key value table metadata store (kvt metadata store). Although KVT  presently only support static partitioning, we have defined the metadata schema which can support scaling in future. This means we maintain KVT Epoch Records which include KVT Segments. All of this is stored in dedicated metadata tables (table segments) per key value table.
 
-Each bucket corresponds to a unique znode in
-Zookeeper. A qualified scoped Stream name is used to compute a
-hash value to assign the Stream to a bucket. All Controller instances, upon
-startup, attempt to take ownership of buckets. Upon _failover_, ownerships
-are transferred, as surviving nodes compete to acquire ownership of
-orphaned buckets. The Controller instance which owns a bucket is
-responsible for all long running scheduled background work corresponding
-to all nodes under the bucket. Presently this entails running periodic
-workflows to capture `StreamCut`(s) (called Retention-Set) for each Stream at desired frequencies.
+- **KVT Epoch Records:**  
+_KVTEpoch: ⟨time, list-of-segments-in-epoch⟩_.
+We store the series of _active_ Table Segments as they transition from one epoch to another. Presently there is only one epoch record for each KVT. 
 
-### Retention Set
+- **KVT Segment Records:**
+_KVT Segment-info: ⟨segmentid, time, keySpace-start, keySpace-end⟩_.
+ Each KVT segment refers to a table segment in segment store. The Controller stores KVT Segment information within each epoch record. The additional metadata about each table segment that controller maintains with KVT Segment includes keySpace-start and keySpace-end which is used for paritioning keys across different KVT segments. 
 
- One retention set per Stream is stored under the corresponding
- bucket/Stream znode. As we compute  `StreamCut`(s) periodically, we keep
- preserving them under this znode. As some automatic truncation is
- performed, the `StreamCut`(s) that are no longer valid are purged from
- this set.
+- **KVT State:**
+_KVT State is an enum that describes various states in the lifecycle of KV table. It ranges from CREATING, ACTIVE, DELETING. 
 
 ## Controller Cluster Listener
```

Comment:
This section should be under `Cluster Management` section, since it essentially deals with managing the Pravega cluster.

Reply:
This section states:
```Each Controller instance has multiple sub components that implement the failover sweeper interface. Presently there are three components that implement failover sweeper interface namely...
```
Before we state this, it may be good to explain what a `failoverSweeper` interface is for and what are the implementing classes expected to do. I understand in a later statement we say that it "sweeps" tasks orphaned by any exiting Controller instance... but a more detailed explanation is required here for users to be able to understand how we handle `Controller fail-overs`

Reply:
> This sub-section could be under Cluster Management section, since it essentially deals with managing the Pravega cluster..... and doing failovers for Controller servers....

Actually these are two different subsystems - 1 is talking about controller cluster management where the failover of task one a failed controller is handled.
while the other talks about managing segment store cluster and distributing segment containers within it.